### ISSUE TITLE
operatorhubio-catalog is scheduled to run on a MS Windows worker node

### ISSUE THREAD

**Issue by HansK-p** [operator-framework/operator-lifecycle-manager#1119]
## Bug Report

**What did you do?**
A clear and concise description of the steps you took (or insert a code snippet).
This is an AKS cluster running K8s 1.14.8 with multiple node pools enabled and both a Linux and a MS Windows Node Pool. The single node in the MS Windows Nodepool is tainted:
<pre>
taints:
  - effect: NoSchedule
    key: os
    value: Win2019
</pre>

The operator has been installed with and unmodified version of the install script, that is:
<pre>
/install.sh 0.12.0
<pre>

What we see is that the pod operatorhubio-catalog-h4zqh is scheduled to run on the MS Windows node, which of course doesn't work.

**What did you expect to see?**
I would expect the pod operatorhubio-catalog-h4zqh to end up on an Linux node. Especially as the MS Windows node is tainted.

**What did you see instead? Under which circumstances?**
The pod operatorhubio-catalog-h4zqh ended up on a MS Windows worker node.

**Environment**
* operator-lifecycle-manager version:

 0.12

* Kubernetes version information:

Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"211047e9a1922595eaa3a1127ed365e9299a6c23", GitTreeState:"clean", BuildDate:"2019-10-15T12:02:12Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: Azure AKS cluster with support for multiple node pools and an active MS Windows node-pool.

**Possible Solution**
Make sure that the deployment of pod  operatorhubio-catalog-h4zqh respect taints and/or a node selector mechanism is used (as it already is for deployments in the same namespace).

**Additional context**

**Comment by Denton24646** [comment]
Hi @HansK-p, thanks for bringing this issue up. This is something we've been discussing this past week as we see more mixed linux and windows OS node pools. We will look into this and provide a fix.

**Comment by Denton24646** [comment]
@HansK-p now that I'm looking at the issue more closely, I'm not entirely sure how the pod was scheduled onto the Windows node in the first place if the pod never tolerated the taint of the node. What you're describing is basically a dedicated node with a specific taint, and only pods which have tolerations which match would be scheduled, per https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/#example-use-cases. 

We can add a `NodeSelector` to the pod, which would help avoid the issue from occurring again (since that label is not present on the node), but it wouldn't explain why the catalog pod was scheduled onto the Windows node in the first place.

**Comment by HansK-p** [comment]
I saw the Exists toleration, but didn't know that it was that "effective". The pod was scheduled on the MS Windows node even after I marked the node as unschedulable (uncordon). This was slightly frustrating....

Adding a NodeSelector to the pod sounds like a good idea. It should never be scheduled to run on anything but a Linux node (unless it actually works).  I assume the NodeSelector will be:
<pre>
  nodeSelector:
    beta.kubernetes.io/os: linux
</pre>
